tasks/hyrax-metadata-updates/index.js

```diff
@@ -118,6 +118,55 @@ async function getEntryTitle(config, metadata, isUmmG) {
   return encodeURIComponent(datasetId);
 }
 
+/**
+ * getCollectionEntry
+ *
+ * @param {Object} config - configuration
+ * @param {Object} metadata - the granule metadata
+ * @param {boolean} isUmmG - whether this is UMM-G or ECHO10 metadata
+ * @returns {Promise<string>} the collection url entry of the collection this granule
+ *    belongs to
+ */
+async function getCollectionEntry(config, metadata, isUmmG) {
+  let conceptId;
+  let shortName;
+  let versionId;
+  if (isUmmG === true) {
+    shortName = metadata.CollectionReference.ShortName;
+    versionId = metadata.CollectionReference.Version;
+  } else {
+    shortName = metadata.Granule.Collection.ShortName;
+    versionId = metadata.Granule.Collection.VersionId;
+  }
+
+  const cmrSettings = await getCmrSettings({
+    ...config.cmr,
+    ...config.launchpad,
+  });
+
+  // Query CMR for collection and retrieve entry title
+  const cmrInstance = new CMR(cmrSettings);
+
+  const searchParams = {
+    short_name: shortName,
+    version: versionId,
+  };
+
+  const result = await cmrInstance.searchCollections(searchParams);
+  // Verify that we have a valid result. If we don't then something is badly wrong and we should halt.
+  // Either the code is faulty or the provider is trying to ingest granules into a collection that doesn't exist
+  conceptId = get(result, '[0].id');
+  shortName = get(result, '[0].short_name');
+  versionId = get(result, '[0].version_id');
+  if (conceptId === undefined) {
+    throw new RecordDoesNotExist(`Unable to query parent collection using short name ${shortName} and version ${versionId}`);
+  }
+
+  return (config.addShortnameAndVersionIdToConceptId !== undefined
+         && config.addShortnameAndVersionIdToConceptId ===true) ?
+         conceptId + '/' + shortName + '.' + versionId : conceptId;
```

Comment:
To ask a dumb question: using `conceptId + '/' + shortName + '.' + versionId` vs `conceptId` in the URL produces an equivalent response from CMR? If so, what is the purpose of adding this functionality to produce the former?

Reply:
https://wiki.earthdata.nasa.gov/display/DUTRAIN/Feature+analysis:+Restified+URL+for+OPENDAP+Data+Access